### PR TITLE
Refactor pinot-batch-ingestion job specs to pinot-spi

### PR DESCRIPTION
**Pull request by fx19880617**
1. Refactor job specs from `pinot-batch-ingestion-base` module to `pinot-spi` module
2. Adding `IngestionJobRunner` interface to be implemented by ingestion jobs.
3. Create a job launcher class `IngestionJobLauncher` in `pinot-spi`
4. Rename `pinot-batch-ingestion-base` to `pinot-batch-ingestion-standalone` and make it a plugin.

### CODE REVIEW

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/standalone/SegmentGenerationJobRunner.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.ingestion.standalone;
+package org.apache.pinot.plugin.ingestion.standalone;
```
Comment (kishoreg) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/standalone/SegmentGenerationJobRunner.java]: plugin.batch....

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/standalone/SegmentGenerationJobRunner.java [hunk 2]
```diff
@@ -36,31 +36,66 @@
...
-import org.apache.pinot.ingestion.common.SegmentGenerationTaskSpec;
-import org.apache.pinot.ingestion.common.PinotClusterSpec;
-import org.apache.pinot.ingestion.common.PinotFSSpec;
-import org.apache.pinot.ingestion.common.SegmentGenerationJobSpec;
 import org.apache.pinot.common.config.TableConfig;
 import org.apache.pinot.common.utils.TarGzCompressionUtils;
 import org.apache.pinot.spi.data.Schema;
 import org.apache.pinot.spi.filesystem.PinotFS;
 import org.apache.pinot.spi.filesystem.PinotFSFactory;
+import org.apache.pinot.spi.ingestion.runner.IngestionJobRunner;
```
Comment (kishoreg) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/standalone/SegmentGenerationJobRunner.java]: spi.batch....

File: pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/spec/Constants.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.ingestion.common;
+package org.apache.pinot.spi.ingestion.spec;
 
 public class Constants {
```
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/spec/Constants.java]: It's been used from all the jobRunners, I think it's fine to keep it here. We may add more constants later.

**Comment by Jackie-Jiang** [comment]
Might be better to change package `org.apache.pinot.spi.batch.ingestion.*` -> `org.apache.pinot.spi.ingestion.batch.*` (reverse batch and ingestion) as we will likely add `org.apache.pinot.spi.ingestion.stream.*` in the future?

**Comment by fx19880617** [comment]
> Might be better to change package `org.apache.pinot.spi.batch.ingestion.*` -> `org.apache.pinot.spi.ingestion.batch.*` (reverse batch and ingestion) as we will likely add `org.apache.pinot.spi.ingestion.stream.*` in the future?

sounds good

**Comment by fx19880617** [comment]
> LGTM. Can you run a reformat (re-order imports) to the changed files before merge?

Thanks for the reminder.

**Pull request merged**